### PR TITLE
Add scripts to configure Apache Spark clusters

### PR DESCRIPTION
**Pull request by daniel-thom**
This PR lets Eagle users create ephemeral Spark clusters on compute nodes and run jobs against them. If the reviewers would like a demo of the functionality or a discussion on why I implemented the functionality this way, please let me know.

### CODE REVIEW

**Comment by tdthatcher** [comment]
We're working on some possibilities to store your image file in an alternate location. Otherwise, this looks good.

**Review by tdthatcher — COMMENTED** [review]
I haven't given it a deep-dive technical review (i.e. I haven't spun up a Spark cluster myself using the instructions/scripts.) I have looked it over for general technical correctness and to be sure there isn't any sensitive information, as well as basic grammar/spelling, and it looks good. This is overall a great contribution to the repository and much appreciated.

**Comment by daniel-thom** [comment]
@tdthatcher I pushed some usability improvements today. There might be room for more. One colleague suggested incorporating environment modules. One possibility is something like this:
```
module load apache-spark/python3.9
```
Where the target is the name of an existing container. While I would like to keep it simple for users to customize their own containers, providing a default option where minimal configuration is required would be good. I'm open to your team's preferences here.

Another item I explored this week: integration with Nvidia GPUs. Nvidia claims that their Rapids plugin can deliver massive improvements in certain types of queries. I actually had this working earlier this year, but that container no longer works. In order to make that work I think I would need help figuring out compatibility between the very old drivers on Eagle and the current Nvidia Spark plugins. I'm also OK leaving that out for now.

**Comment by yandthj** [comment]
The image file has been moved out of /scratch to /datasets. Daniel is having a colleague test this out, so we'll hold off on merging until he confirms that is done.

**Comment by daniel-thom** [comment]
> The image file has been moved out of /scratch to /datasets. Daniel is having a colleague test this out, so we'll hold off on merging until he confirms that is done.

@yandthj This should be ready to merge now.

File: applications/spark/spark_scripts/configure_spark.sh [hunk 1]
```diff
@@ -0,0 +1,240 @@
...
+CONFIG_DIR=$(pwd)
+DRIVER_MEMORY_GB=1
+ENABLE_DYNAMIC_ALLOCATION=false
+ENABLE_HISTORY_SERVER=false
+# Many online docs say executors max out with 5 threads.
+EXECUTOR_CORES=5
+PARTITION_MULTIPLIER=1
+SLURM_JOB_IDS=()
+
+function check_errors()
```
Comment (k1nshuk) [inline applications/spark/spark_scripts/configure_spark.sh:12]: The functions in this file could benefit from comments/docs that describe what each function does.

File: applications/spark/tests/test_job.py [hunk 2]
```diff
@@ -0,0 +1,24 @@
+import shutil
+from pathlib import Path
```
Comment (k1nshuk) [inline applications/spark/tests/test_job.py:2]: Remove unused imports

**Review by k1nshuk — APPROVED** [review]
Overall looks great to me. I had a couple of minor comments, but this can be merged without including.

File: applications/spark/tests/test_job.py (hunk 2 above)
Comment (daniel-thom) [reply applications/spark/tests/test_job.py]: Thanks for noticing. Fixed.

**Comment by yandthj** [comment]
@daniel-thom Is this ready to be merged in?

**Comment by daniel-thom** [comment]
> @daniel-thom Is this ready to be merged in?

@yandthj I pushed two minor commits just now related to some user feedback I received yesterday. I think it's ready to merge. It's possible that users will want more documentation improvements, but that can be done later.

**Review by yandthj — APPROVED** [review]

**Pull request merged**